src/NLog/Targets/Wrappers/AsyncTargetWrapper.cs

```diff
@@ -173,10 +181,26 @@ public int QueueLimit
         [DefaultValue(5)]
         public int FullBatchSizeWriteLimit { get; set; }
 
+#if NET4_5 || NET4_0
+        /// <summary>
+        /// Gets or sets whether to use the locking queue, instead of a lock-free concurrent queue
+        /// The locking queue is less concurrent when many logger threads, but reduces memory allocation
+        /// </summary>
+        public bool ForceLockingQueue
+        {
+            get { return _requestQueue is AsyncRequestQueue; }
+            set
+            {
+                if (value != (_requestQueue is AsyncRequestQueue))
```

Comment:
I would prefer having a backend boolean for this (e.g. _forceLockingQueue), and and reset _requestQueue  if  `value != _forceLockingQueue` . 

That would be more logical IMO. What do you think?